rust_snuba/src/processors/errors.rs

```diff
@@ -183,6 +185,26 @@ struct TraceContext {
     other: GenericContext,
 }
 
+#[derive(Debug, Deserialize, JsonSchema)]
+#[serde(untagged)]
+enum FeatureContextEnum {
+    Typed(FeatureContext),
+    #[allow(warnings)]
+    Untyped(Value),
```

Comment:
I don't think this is a reasonable schema. every time there is any schema incompatibility within the feature context, we drop it entirely. why are there sometimes untyped values in this context?

why is `allow(warnings)` needed?

Reply:
At least one customer sends their feature flags on the context using the same key as us but not the same format.  So I didn't want to fail de-serialization for them.  Allow warnings is used because we never access the inner value and it produces a lint error.

Reply:
I suggest to handle this case in Relay. It would null out or rename the context, and already has a lot of facilities for partial deserialization/partial errors.

Reply:
@untitaker Whats the purpose of doing this in Relay? Why should we not deserialize in this way in Snuba?

Reply:
think of not only snuba as downstream consumers of the ingestion topic, but also sbc or the UI. they'd all have to handle these compatibility issues if relay didn't iron them out at the start.

i suggest to either:

* drop user-defined contexts if they don't match the schema in relay
* rename them within relay to another key

either way you will be able to have a stricter schema in the ingestion topic and then in snuba, and the complexity for handling those different user inputs is contained in relay instead of {sbc, snuba, ...}